src/Store.js

```diff
@@ -150,7 +160,60 @@ class Store {
   }
 
   /**
-   * finds an instance by `id`. If available in the store, returns that instance. Otherwise, triggers a fetch.
+   * Gets a record from the store, will not fetch from the server if it doesn't exist in store.
+   * If given queryParams, it will check the cache for the record.
+   *
+   * @method getOne
+   * @param {String} type the type to find
+   * @param {String} id the id of the record to get
+   * @param {Object} options { queryParams }
+   */
+  getOne = (type, id, options = {}) => {
+    const { queryParams } = options
+    if (queryParams) {
+      return this.getCachedRecord(type, id, queryParams)
+    } else {
+      return this.getRecord(type, id)
+    }
+  }
+
+  /**
+   * Fetches record by `id` from the server and returns a Promise.
+   *
+   * @async
+   * @method fetchOne
+   * @param {String} type the record type to fetch
+   * @param {String} id the id of the record to fetch
+   * @param {Object} options { queryParams }
+   * @return {Object} record
+   */
+  async fetchOne (type, id, options = {}) {
+    const { queryParams } = options
+    const url = this.fetchUrl(type, queryParams, id)
+    const response = await this.fetch(url, { method: 'GET' })
+
+    if (response.status === 200) {
+      const json = await response.json()
+      const { data, included } = json
+
+      if (included) {
+        this.createModelsFromData(included)
+      }
+
+      const record = this.createOrUpdateModel(data)
+
+      this.data[type].cache.set(url, [record.id])
+      // TODO:  return Promise.resolve(record)
```

Comment:
do we need `Promise.resolve`?

Reply:
Ah it's an async method so maybe not so much thanks